CMakeLists.txt

```diff
@@ -0,0 +1,1045 @@
+# Update the documentation string of CMAKE_BUILD_TYPE for GUIs to include the MPI options
+SET(CMAKE_BUILD_TYPE Release CACHE STRING
+#SET(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}" CACHE STRING
+	"Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel MPI MPIDEBUG."
+	FORCE)
+
+###################################################################################################
+# Sixtrack configuration options
+###################################################################################################
+
+# Now we have the default settings
+# we have ON
+OPTION(TILT "Allow elements to be tilted (by error)" ON)
+OPTION(FAST "Which implementation of drifts to use in thin6d" ON)
+OPTION(CRLIBM "Use correctly rounded libmath instead of system libmath" ON)
+OPTION(CERNLIB "Link to external CERNLIB library for PAW plots. Otherwise use internally defined dummy functions from deck plotdumy" OFF)
+OPTION(NAGLIB "Link to external NAGLIB library, used in subroutine coruglo and coruord for DA version" OFF)
+# used for ifort only SSE
+#SSE="NULL"
+# and we have OFF
+OPTION(API "Link to external BOINC API" OFF)
+OPTION(DA "Build differential algebra version (NOT dynamic aperture!)" OFF)
+OPTION(COLLIMAT "Collimation code" OFF)
+OPTION(CPSS "CERN physics screensaver support" OFF)
+OPTION(BOINC "BOINC distributed computing support" OFF)
+OPTION(CR "Checkpoint / restarting" OFF)
+OPTION(BPM "Activates the collection of tracking data for sussix (deprecated)" OFF)
+OPTION(BEAMGAS "Beam-gas scattering" OFF)
+OPTION(BNLELENS "Support for the BNL electron lens (deprecated)" OFF)
+OPTION(BIGNBLZ "Allow up to 200'000 structure elements (normally 20'000 (15k for collimat)) - useful for FCC etc." OFF)
+OPTION(HUGENBLZ "Allow up to 400'000 structure elements (normally 20'000 (15k for collimat)) - useful for FCC etc." OFF)
+OPTION(STF "Single Track File - write all tracks for postprocessing to fort.90, instead of 32 separate files fort.90 to fort.(90-32). Required for increased particle numbers" OFF)
+OPTION(BIGNPART "Allow tracking up to 2048 particles (normally 64)." OFF)
+OPTION(HUGENPART "Allow tracking up to 65536 particles (normally 64)" OFF)
+OPTION(DATAMODS "Allocate big matrices for thick tracking on the heap and only if required" OFF)
+OPTION(DEBUG "Debugging output" OFF)
+OPTION(HDF5 "Link to external HDF5 library, write tracks2.dat data (for postprocessing with BeamLossPattern) there instead of writing it to ASCII." OFF)
+OPTION(FIO "Use FortranIO from Fortran2003 to correctly round ASCII input/output" OFF)
+OPTION(MERLINSCATTER "Interaction physics for collimation from Merlin" OFF)
+
+#Compiler stuff
+OPTION(IFORT "Use the intel fortran compiler" OFF)
+OPTION(LF95 "Use the Lahey-Fujitsu lf95 fortran compiler" OFF)
+OPTION(PGF90 "Use the pgf90 fortran compiler" OFF)
+OPTION(F95 "Use the g95 fortran compiler" OFF)
+OPTION(GFORTRAN "Use the gfortran fortran compiler" OFF)
+OPTION(NAGFOR "Use NAGFOR compiler" OFF)
+
+#Build options
```

Comment:
Maybe comment out the irrelevant build options like OPENMP, MPI, CUDA, ROOT until it becomes relevant?